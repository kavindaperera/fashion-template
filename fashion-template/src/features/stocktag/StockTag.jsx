import React from "react";
import { Label, Icon } from "semantic-ui-react";
function StockTag({ stock, selectedVariant }) {
  return (
    <div>
    {stock!= null ? (
        <div>
        {stock > 0 ? (
          <Label size="big" color="green">
            <Icon name="check" /> {stock + " items in stock"}
          </Label>
        ) : (
          <Label size="big" color="red">
            <Icon name="ban" /> {"Out of Stock"}
          </Label>
        )}
      </div>
    ): ( <p></p>)
    }
    </div>
  );
}

export default StockTag;
