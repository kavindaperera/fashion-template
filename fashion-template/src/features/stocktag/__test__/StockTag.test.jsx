import React from 'react';
import ReactDOM from 'react-dom';
import StockTag from '../StockTag'
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'


afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<StockTag></StockTag>,div)
});

it("renders stock tag correctly",()=>{
    const {getByTestId} = render(<StockTag enableInventoryManagement={true} stock={10}></StockTag>)
    expect(getByTestId('stock-tag')).toHaveTextContent("10 items in stock")
});

it("renders stock tag correctly",()=>{
    const {getByTestId} = render(<StockTag stock={0}></StockTag>)
    expect(getByTestId('stock-tag')).toHaveTextContent("Out of Stock")
});

it("renders stock tag correctly",()=>{
    const {getByTestId} = render(<StockTag stock={null}></StockTag>)
    expect(getByTestId('stock-tag')).toHaveTextContent("")
});

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<StockTag stock={10}></StockTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<StockTag stock={0}></StockTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<StockTag stock={null}></StockTag>).toJSON();
    expect(tree).toMatchSnapshot();
})