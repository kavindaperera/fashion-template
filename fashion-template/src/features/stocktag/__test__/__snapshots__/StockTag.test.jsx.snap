// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="stock-tag"
>
  <div />
</div>
`;

exports[`matches snapshot 1 2`] = `
<div
  data-testid="stock-tag"
>
  <div>
    <div
      className="ui red small basic label"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="ban icon"
      />
       
      Out of Stock
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1 3`] = `
<div
  data-testid="stock-tag"
>
  
</div>
`;
